clr-k8s-examples/setup_system.sh

```diff
@@ -3,85 +3,159 @@
+  # add localhost to /etc/hosts file
+  hostcount=$(grep '127.0.0.1 localhost' /etc/hosts | wc -l)
+  if [ "$hostcount" == "0" ]; then
+    echo "127.0.0.1 localhost $(hostname)" | sudo bash -c "cat >> /etc/hosts"
+  else
+    echo "/etc/hosts already configured"
+  fi
+}
 
-sudo systemctl daemon-reload
-# This will fail at this point, but puts it into a retry loop that
-# will therefore startup later once we have configured with kubeadm.
-echo "The following kubelet command may complain... it is not an error"
-sudo systemctl enable --now kubelet $RUNNER || true
+# daemon reload
+function daemon_reload() {
+  sudo systemctl daemon-reload
+}
+
+# enable kubelet for $RUNNER
+function enable_kubelet_runner() {
+  # This will fail at this point, but puts it into a retry loop that
+  # will therefore startup later once we have configured with kubeadm.
+  echo "The following kubelet command may complain... it is not an error"
+  sudo systemctl enable --now kubelet $RUNNER || true
+}
 
-#Ensure that the system is ready without requiring a reboot
-sudo swapoff -a
-sudo systemctl restart systemd-modules-load.service
+# ensure that the system is ready without requiring a reboot
+function ensure_system_ready() {
+  sudo swapoff -a
+  sudo systemctl restart systemd-modules-load.service
+}
 
+# add proxy if found
+function setup_proxy(){
 set +o nounset
 if [[ ${http_proxy} ]] || [[ ${HTTP_PROXY} ]]; then
-	echo "Setting up proxy stuff...."
-	# Setup IP for users too
-	sed_val=${ADD_NO_PROXY//\//\\/}
-	[ -f /etc/environment ] && sudo sed -i "/no_proxy/I s/$/,${sed_val}/g" /etc/environment
-	if [ -f /etc/profile.d/proxy.sh ]; then
-		sudo sed -i "/no_proxy/I s/\"$/,${sed_val}\"/g" /etc/profile.d/proxy.sh
-	else
-		echo "Warning, failed to find /etc/profile.d/proxy.sh to edit no_proxy line"
-	fi
-
-	services=($RUNNER 'kubelet')
-	for s in "${services[@]}"; do
-		sudo mkdir -p "/etc/systemd/system/${s}.service.d/"
-		cat <<EOF | sudo bash -c "cat > /etc/systemd/system/${s}.service.d/proxy.conf"
+  echo "Setting up proxy stuff...."
+  # Setup IP for users too
+  sed_val=${ADD_NO_PROXY//\//\\/}
+  [ -f /etc/environment ] && sudo sed -i "/no_proxy/I s/$/,${sed_val}/g" /etc/environment
+  if [ -f /etc/profile.d/proxy.sh ]; then
+    sudo sed -i "/no_proxy/I s/\"$/,${sed_val}\"/g" /etc/profile.d/proxy.sh
+  else
+    echo "Warning, failed to find /etc/profile.d/proxy.sh to edit no_proxy line"
+  fi
+
+  services=($RUNNER 'kubelet')
+  for s in "${services[@]}"; do
+    sudo mkdir -p "/etc/systemd/system/${s}.service.d/"
+    cat <<EOF | sudo bash -c "cat > /etc/systemd/system/${s}.service.d/proxy.conf"
 [Service]
 Environment="HTTP_PROXY=${http_proxy}"
 Environment="HTTPS_PROXY=${https_proxy}"
 Environment="SOCKS_PROXY=${socks_proxy}"
 Environment="NO_PROXY=${no_proxy},${ADD_NO_PROXY}"
 EOF
-	done
+  done
 fi
 set -o nounset
+}
+
+# init for performing any pre tasks
+function init(){
+  echo ""
+}
+
+###
+# main
+##
+set +o nounset
+if [[ -n "${CLRK8S_OS}" ]]; then
```

Comment:
Won't this fail since we are not setting the default value for this?

Reply:
The `set +o nounset` allows it to not exist. This is set back right after with `set -o nounset`.

Reply:
Instead can we initialize it as an empty string at the beginning and check with `-z`?

Reply:
I used `-n` (equivalent to `! -z`) but defaulted to "" and took out the `nounset`s.